corehq/motech/auth.py

```diff
@@ -274,6 +276,19 @@ def set_last_token(token):
         )
         return session
 
+    def _check_last_token(self, domain_name):
```

Comment:
This is annoying. The [deposit V2 API](https://momodeveloper.mtn.com/api-details#api=disbursement&operation=Deposit-V2) specifies the following for the Authorization header:
_"Bearer Authentication Token generated using CreateAccessToken API Call"_

but the [CreateAccessToken API](https://momodeveloper.mtn.com/api-details#api=disbursement&operation=CreateAccessToken) returns the following:
```
{
    "access_token": <a valid access token>,
    "expires_in": 3600,
    "token_type": "access_token"
}
```
The `OAuth2Session` expects the `self.last_token` to have the `token_type` set to `"Bearer"` and not `"access_token"`.

I reached out to support and am awaiting their response, but in the meantime this workaround will have to do I guess.

Reply:
I'm not sure if this will be the case for the live API as well, but I know this is true for the sandbox.

Reply:
Hey @Charl1996 

Can you try this?

I think you need to use the token returned as the access token to fetch a Bearer token from https://momodeveloper.mtn.com/api-details#api=disbursement&operation=CreateOauth2Token and then send the request.

Reply:
I can try that, but if I use the `access_token` from the CreateAccessToken response I can successfully request payment (which should not work if it had to be a token from the CreateOauth2Token endpoint, right?)

Reply:
Ok, after quite a struggle I've managed to get an oauth access token and tried the [Deposit-V2 endpoint](https://momodeveloper.mtn.com/api-details#api=disbursement&operation=Deposit-V2) with the oauth token, but I get a 401 status code as opposed to the access token retrieved the [CreateAccessToken endpoint](https://momodeveloper.mtn.com/api-details#api=disbursement&operation=CreateAccessToken) giving a 202 (as expected).

My conclusion is that the CreateAccessToken is the right way to authenticate.

Reply:
Okay, thanks for giving it a go.
I see this whole integration has been quite a struggle so happy to put this down for now.